api/cmd/build/main.go

```diff
@@ -123,22 +123,43 @@ func cloneGit(s string) {
 		os.Setenv("GIT_SSH_COMMAND", "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no")
 	}
 
+	err = ioutil.WriteFile("/root/.netrc", []byte(fmt.Sprintf("machine %s login %s", u.Host, u.User.Username())), 0700)
+	if err != nil {
+		fmt.Println("WARNING: Failed to write to .netrc; git might fail: ", err)
+	}
+
 	writeAsset("/usr/local/bin/git-restore-mtime", "git-restore-mtime", 0755, nil)
 
 	run(".", "git", "clone", "--progress", repo, "src")
```

Comment:
If the .netrc strategy works, you should be able to remove u.User from the repo argument, which would fix https://github.com/convox/rack/issues/833